File: gherkin/java/src/main/java/gherkin/ast/DataTable.java
package gherkin.ast;

import java.util.Collections;
import java.util.List;

public class DataTable extends Node {
    private final List<TableRow> rows;

    public DataTable(List<TableRow> rows) {
        super(rows.get(0).getLocation());
        this.rows = Collections.unmodifiableList(rows);
    }

    public List<TableRow> getRows() {
        return rows;
    }
}

File: gherkin/dotnet/Gherkin/Ast/DataTable.cs
using System;
using System.Collections.Generic;

namespace Gherkin.Ast
{
    public class DataTable : StepArgument, IHasRows, IHasLocation
    {
        public Location Location { get; private set; }
        public IEnumerable<TableRow> Rows { get; private set; }

        public DataTable(TableRow[] rows)
        {
            if (rows == null) throw new ArgumentNullException("rows");
            if (rows.Length == 0) throw new ArgumentException("DataTable must have at least one row", "rows");

            Location = rows[0].Location;
            Rows = rows;
        }
    }
}